### ISSUE TITLE
feature request: support helmfile/vals as a plugin in container

### ISSUE THREAD

**Issue by LPLabrys99** [crumbhole/argocd-lovely-plugin#611]
Hi, I've been using the argocd lovely plugin for a little while now and I think it's really great.
There is one thing that I think could be better - it would be nice if the container supported the helmfile/vals project by default.
https://github.com/helmfile/vals

Vals is a secrets connector that supports various backends out of the box. As a kubernetes operator you can specify references to secrets in your values files with a format like `ref+gcp://....` and vals will look up the secret when templating the charts. It comes built into helmfile but can also be run as a standalone app. Helmfile comes with vals built-in to its own binary but there doesn't seem to be a way to invoke vals through helmfile. If you want to use vals by itself you have to install it as a standalone app.

As a user of argocd/lovely plugin, if I am deploying a single simple Helm chart with some secrets and don't want to set up another helmfile in the repo it would be ideal to use argocd + lovely plugin with standard helm + vals in the pipeline.

**Comment by Joibel** [comment]
It looks like this would be achievable with a lovely [variation](https://github.com/crumbhole/argocd-lovely-plugin/blob/main/doc/variations.md)

Would you like to contribute a PR for this?

**Comment by LPLabrys99** [comment]
I don't have experience in golang but I'm willing to try.

By the way, given that lovely supports pipeline customisation, is there a reason why vals shouldn't be included in all variations as helmfile is currently?

**Comment by Joibel** [comment]
Adding vals: The same argument could be applied to any extra tool that manipulates text.

Vals is not a core feature of lovely. Variations were created specifically for people who want extra tools without adding further possibly security issues to the core of lovely by putting all the tools in the core docker image.

Adding a variation mostly involves writing a Dockerfile, not golang, so perhaps you can tackle it?

**Comment by rubentest** [comment]
we have any update of this?

**Comment by Joibel** [comment]
No. If you'd like to do a PR please do - as a variation I would accept it.
